pkg/hive/install.go

```diff
@@ -167,6 +169,11 @@ func (c *clusterManager) clusterDeploymentForInstall(doc *api.OpenShiftClusterDo
 		}
 	}
 
+	clusterDomain := doc.OpenShiftCluster.Properties.ClusterProfile.Domain
+	if !strings.ContainsRune(clusterDomain, '.') {
+		clusterDomain += "." + os.Getenv("DOMAIN_NAME")
+	}
+
```

Comment:
This bit of code is in our code base so many times. It would be great if we could extract it into a function and apply some testing to the result (like a regex match to ensure it's a properly formed FQDN).

Reply:
Definitely makes sense, but we should do so in a follow up PR.

Reply:
[Created a WI](https://dev.azure.com/msazure/AzureRedHatOpenShift/_workitems/edit/16616511) for follow-up work.